packages/iris-grid/src/CommonTypes.tsx

```diff
@@ -59,7 +59,12 @@ export type InputFilter = {
 };
 
 export interface UIRow {
-  data: Map<ModelIndex, CellData>;
+  /**
+   * The data in the row indexed by column number.
+   * If a column is not part of the columns array (i.e. it's hidden by the model/table),
+   * then it will be included by its name instead of index.
+   */
+  data: Map<ModelIndex | ColumnName, CellData>;
```

Comment:
This shouldn't be necessary? I think it should map to the `dataBarOptionsForCell` already no?

Reply:
No. This is part of `model.viewportData`.

The problem is when we subscribe to columns and the UITableModel in plugins adds the auto-agg columns to the subscription, but hides them from `model.columns`. In the `extractViewportRow` method, we expect the column to exist in `model.columns` because we index based on our model index.

For the auto-agg columns which are hidden from `model.columns`, but added to the subscription, these show up in `extractViewportRow`, but don't have a model index associated with the column.

So this change is so subscribing to columns that aren't part of `model.columns` works without throwing.

The value is then read by column name from the model `viewportData` in the plugin

Reply:
Ah yea - `extractViewportRow` should be using `this.table.columns` instead of `this.columns`, then I think it works out.
The viewport data should only be accessed internally from that model, so the access should always match up with the table columns I believe.

Reply:
I'll try moving the proxy off of the table and onto the model in the plugin for `columns`. `model.columns` just returns `this.table.columns` by default.

My concern with that was somewhere not using `model.columns` and instead using `table.columns` and causing some breakage

Just glancing through `IrisGridTableModelTemplate`, `get columnMap` accesses `this.table.columns` instead of `this.columns`. It's something that's hard to guard against

Reply:
And using `this.table.columns` in `extractViewportRow` might not fix it. It already iterates the columns in the viewport data and then we lookup the index with `this.getColumnIndexByName`. So if the model hides the column from `this.columns`, then it still won't have an index for aggregations when extracting the viewport row.

Reply:
So do you want me to change anything here? I can get rid of `get columnMap`

I think the proper way (as a separate ticket) would be to move away from the notion of a model index and just to names or visible index. We already do this for some things (like sorts seem to be serialized by name), so I'm not sure where model index should be used over the column name.